src/backend/api/meals.js

```diff
@@ -1,12 +1,79 @@
 const express = require("express");
+const { request, response } = require("../app");
 const router = express.Router();
 const knex = require("../database");
 
+// maxPrice
 router.get("/", async (request, response) => {
+  let meals = knex("meal");
+  if ("maxPrice" in request.query) {
+    if (!"maxPrice") {
+      response.status(403).json("Please enter data");
+    } else {
+      meals = meals.where("price", "<=", Number(request.query.maxPrice));
+    }
+  }
+  // availableReservations
+
+  if ("availableReservation" in request.query) {
+    const reservation = request.query.availableReservation;
+    if (false(reservation) && typeof reservation == "boolean") {
+      response.status("Sorry there are no reservations!");
+    } else if (true(reservation) && typeof reservation == "boolean") {
+      meals = meals
+        .join("reservation", "meal.id", "=", "reservation.meal_id")
+        .select(
+          "meal.id",
+          "title",
+          "max_reservations",
```

Comment:
`meal. title, meal.max_reservations`  would be great to avoid ambiguity since there is a column called  'title' in the reservations table and of course for the sake of consistency since you are using "meal.id".